perfscale_regression_ci/scripts/encryption/toggle_encryption.sh

```diff
@@ -0,0 +1,151 @@
+((oc_secrets_num = oc_secrets_num + 1))
+((kb_secrets_num = kb_secrets_num + 1))
+((oauth_secrets_num = oauth_secrets_num + 1))
+oc_secrets_name=$(echo -e "encryption-key-openshift-apiserver-${oc_secrets_num}")
+kb_secrets_name=$(echo -e "encryption-key-openshift-kube-apiserver-${kb_secrets_num}")
+oauth_secrets_name=$(echo -e "encryption-key-openshift-oauth-apiserver-${oauth_secrets_num}")
+log "Secrets to check:"
+log "${oc_secrets_name}"
+log "${kb_secrets_name}"
+log "${oauth_secrets_name}"
+
+function enable_encryption {
+    mode="aescbc"
+    oc patch apiserver/cluster -p '{"spec":{"encryption": {"type":"aescbc"}}}' --type merge
+    log "Encryption enabled"
+}
+
+function disable_encryption {
+    mode="identity"
+    oc patch apiserver/cluster -p '{"spec":{"encryption": {"type":"identity"}}}' --type merge
+    log "Encryption disabled"
+}
+
+function check_secret {
+    secret_file="$1"
+    migrated_resources_value_a="$2"
+    migrated_resources_value_b="$3"
+    secret_migrated_resources=$(grep "encryption.apiserver.operator.openshift.io/migrated-resources" "${secret_file}" | awk '{print $2}')
+    secret_timestamp=$(grep "encryption.apiserver.operator.openshift.io/migrated-timestamp" "${secret_file}" | awk '{print $2}')
+    secret_mode=$(grep "encryption.apiserver.operator.openshift.io/mode" "${secret_file}" | awk '{print $2}')
+
+    if [[ ${secret_migrated_resources} == "${migrated_resources_value_a}" ]] || [[ ${secret_migrated_resources} == "${migrated_resources_value_b}" ]]; then
+        log "Founded: encryption.apiserver.operator.openshift.io/migrated-resources in ${secret_file} secret."
+    else
+        log "Missing: encryption.apiserver.operator.openshift.io/migrated-resources in ${secret_file} secret."
+        continue_wait=true
+    fi
+
+    if [[ ${secret_timestamp} != "" ]]; then
+        log "Founded: encryption.apiserver.operator.openshift.io/migrated-timestamp in ${secret_file} secret."
+    else
+        log "Missing: encryption.apiserver.operator.openshift.io/migrated-timestamp in ${secret_file} secret."
+        continue_wait=true
+    fi
+
+    if [[ ${secret_mode} == "${mode}" ]]; then
+        log "Founded: encryption.apiserver.operator.openshift.io/mode in ${kb_secrets_file} secret."
+    else
+        log "Missing: encryption.apiserver.operator.openshift.io/mode in ${kb_secrets_file} secret."
+        continue_wait=true
+    fi
+}
+
+function wait_until_encryption_is_ready {
+    start_time=$(date +%s)
+    while ((($(date +%s) - start_time) < timeout)); do
+        continue_wait=false
+
+        echo -e "Not ready yet. Time from beginning: $(($(date +%s) - start_time)) seconds"
+        echo -e "Retrying in one minute"
+        sleep 60
+
+        # Getting secrets
+        log "Getting ${oc_secrets_name} secret"
+        oc get secret "${oc_secrets_name}" -o yaml -n openshift-config-managed >"${OC_SECRET}"
+        log "${OC_SECRET}"
+        cat "${OC_SECRET}"
```

Comment:
Should we be printing off these secrets? Do we need this step?

Reply:
This is only for debuging. Maybe show this secrets only on failure at the end?